src/viam/rpc/server.py

```diff
@@ -112,7 +113,14 @@ async def serve(
             else:
                 await self._server.start(host, port)
                 LOGGER.info(f"Serving on {host}:{port}")
-            await self._server.wait_closed()
+
+            # This has to be done for a bug in Python 3.12, where logger handlers hang on shutdown
+            task = asyncio.create_task(self._server.wait_closed())
+            if self._server._server_closed_fut is not None:
+                await self._server._server_closed_fut
+            logging.shutdown()
+            await task
+
             await self.close()
```

Comment:
With this new change, do we reach this code now?

Reply:
Doesn't seem like it, no.